PT-2375 - fixed pt-table-sync for tables with generated columns

hpoettker opened the pull request:
The PR is intended to resolve this issue: https://perconadev.atlassian.net/browse/PT-2375

The tests added with `pt-2375.t` both fail with the current code base when executing the generated DML statements. With the proposed change, they run successfully. They test that both `REPLACE` and `UPDATE` statements are generated correctly by `pt-table-sync` when applied to tables with generated columns.

The code is my own creation and it can be distributed under the GPL2 licence.

- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [x] util/update-modules has been ran
- [x] Documentation updated
- [x] Test suite update

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=860) <br/>All committers have signed the CLA.

hpoettker commented:
I updated the PR such that the change is also included in `lib/ChangeHandler.pm`. `pt-table-sync` is the only affected tool by this change.

hpoettker committed:
PT-2375 - fixed pt-table-sync for tables with generated columns

Previously, pt-table-sync generated DML statements that included
the generated columns of a table, which is however rejected by
the database and not necessary for the required sync statements.

hpoettker commented:
Thanks so much for the massive effort to adjust the code base to MySQL 8.4!

I've rebased the changes on the latest commit in 3.x.
